pkg/receiver/gitproviderreceiver/metadata.yaml

```diff
@@ -73,3 +73,17 @@ metrics:
         gauge:
             value_type: int
         attributes: [repository.name, branch.name]
+    git.repository.pull_request.approval.time:
+        enabled: true
+        description: Time for the PR to be approved
+        unit: 1
+        gauge:
+            value_type: int
+        attributes: [repository.name, branch.name]
+    git.repository.pull_request.deployment.time:
```

Comment:
I don't think merge time can be considered deployment time. Can we convert this to merged time & reflect that it is the time of which the PR has been merged?

Reply:
So this isnt saying that the time of merge is the time of deploy, in the query it uses the merge commit of the pr to see if there are any connected deployments and uses the time of that deployment here.  Also since we are basing the pr opened time off the merge time, wouldnt that effectively be merged time as well?

Reply:
what's the data look like from gh's perspective & what do we they count as a deployment

Reply:
I havn't found a 100% clear answer to what is generically consider a deployment, but it seems to be whenever certain tooling calls out to a 3rd party from your repo and does something with it.  So any creating anything with aws,azure,terraform etc generates a deployment and the data in the REST is shown in the [docs](https://docs.github.com/en/free-pro-team@latest/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment) and the graphQL data mirrors most of it.

Reply:
i would love to pair on this, but i won't be able to today. can you see what a deployment looks like in the api for this specific repo?

Reply:
So there arent't actually any deployments for this repo so its all null, for sk8s however it looks like this:
`"deployments": {
                "nodes": [
                  {
                    "createdAt": "2023-04-25T17:53:02Z",
                    "creator": {
                      "login": "rhoofard"
                    },
                    "environment": "AWS",
                    "latestEnvironment": "AWS",
                    "originalEnvironment": "AWS",
                    "payload": null,
                    "state": "INACTIVE",
                    "task": "deploy",
                    "updatedAt": "2023-04-25T18:34:11Z",
                    "description": null
                  }
                ],
                "totalCount": 1
              }`
There's more stuff available than this but its all just redundant info like repository, ref, commit, etc

Reply:
how does sk8s deploy?
i'm available to pair on this today